docs/styles/spec/search-spec.less

```diff
@@ -0,0 +1,38 @@
+@import (reference) "components/form/index";
+@import (reference) "components/search/index";
+
+.search-spec {
+  hx-search-control {
+    /* --------------- Pristine --------------- */
+    > input[type="search"].mock-focus ~ hx-search {
+      .SearchControl(pristine-focus);
+    }
+
+    /* --------------- Disabled --------------- */
+    > input[type="search"]:disabled ~ hx-search {
+      .SearchControl(disabled);
+
+      &.mock-focus ~ hx-search {
+        .SearchControl(disabled-focus);
+      }
+    }
+
+    /* --------------- Changed / Touched --------------- */
+    &[hx-dirty] {
+      > input[type="search"]:invalid.mock-focus ~ hx-search {
+        .SearchControl(invalid-focus);
+      }
+    }
+
+    /* --------------- Styled Invalid --------------- */
+    &.hxInvalid > input[type="search"]:enabled.mock-focus ~ hx-search {
+      .SearchControl(invalid-focus);
+    }
+
+    /* --------------- Mock Subfocus --------------- */
+    > input[type="search"].mock-focus ~ button.mock-subfocus {
```

Comment:
Since these are mock classes for testing, I used `.mock-subfocus` to differentiate between the primary `.mock-focus` and sub-focus styles.

Thoughts?